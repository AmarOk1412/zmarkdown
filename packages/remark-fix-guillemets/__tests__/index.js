import dedent from 'dedent'
import unified from 'unified'
import reParse from 'remark-parse'
import stringify from 'rehype-stringify'
import remark2rehype from 'remark-rehype'

import remarkFixGuillemets from '../src/'
import textrGuillemets from 'typographic-guillemets'
import remarkTextr from '../remark-textr'


const render = (text, config) => unified()
  .use(reParse, {
    gfm: true,
    commonmark: false,
    yaml: false,
    footnotes: true,
    /* sets list of known blocks to nothing, otherwise <h3>hey</h3> would become
    &#x3C;h3>hey&#x3C;/h3> instead of <p>&#x3C;h3>hey&#x3C;/h3></p> */
    blocks: [],
  })
  .use(remarkFixGuillemets)
  .use(remark2rehype)
  .use(stringify)
  .use(remarkTextr, {
    plugins: [
      textrGuillemets,
    ],
    options: {
      locale: 'fr',
    },
  })
  .processSync(text)

test('issue-80', () => {
  const { contents } = render(dedent`
    <<html>> <<html1>> <<html2>> <<html3>> <<html4>> <<**bold**>>
  `)
  expect(contents).toMatchSnapshot()
})

test('no-html-block', () => {
  const { contents } = render(dedent`
    << 1 >>
  `)
  expect(contents).toMatchSnapshot()
})

test('do-not-replace', () => {
  const { contents } = render(dedent`
    <a>>
  `)
  expect(contents).toMatchSnapshot()
})
