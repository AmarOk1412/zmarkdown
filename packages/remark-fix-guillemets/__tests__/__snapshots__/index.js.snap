// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`issue-80 1`] = `"<p>«html» «html1» «html2» «html3» «html4» «<strong>bold</strong>»</p>"`;
